packages/paste-core/components/menu/src/index.tsx

```diff
@@ -1,12 +1,33 @@
-export * from "./useMenuState";
-export * from "./Menu";
-export * from "./MenuButton";
-export * from "./MenuBadge";
-export * from "./MenuGroup";
-export * from "./MenuItem";
-export * from "./MenuItem.styles";
-export * from "./MenuItemCheckbox";
-export * from "./MenuItemRadio";
-export * from "./MenuSeparator";
-export * from "./SubMenuButton";
-export * from "./types";
+import type { MenuInitialState, MenuStateReturn } from "./useMenuState";
+
+// for the sake of documenting the types we rename the state hook types to append Props to the name, so we can docuemnt them and not cause a breaking change.
+type MenuInitialStateProps = MenuInitialState;
+type MenuStateReturnProps = MenuStateReturn;
+
+export type { MenuInitialStateProps, MenuStateReturnProps };
+
+export { useMenuState } from "./useMenuState";
+export type { MenuInitialState, MenuStateReturn } from "./useMenuState";
+export { Menu } from "./Menu";
+export { MenuButton } from "./MenuButton";
+export type { MenuButtonProps } from "./MenuButton";
+export { MenuBadgeContent, MenuBadge } from "./MenuBadge";
```

Comment:
suggestion: I don't think `MenuBadgeContent` is a real component that needs documenting

Reply:
Update the status package only needs MenuBadgeContent (not MenuBadgeContentProps) so I removed the types export